### PR TITLE
Create user model

### PR DESCRIPTION
**Pull request by ronaldemanuel**
# Main changes
* Created the user model with the main fields:
    * User name
    * Email
    * Password
* Implemented migrations for user table
* Implemented 3 user seeders
## Aditional changes
* Added encryption to user seeder passwords

### CODE REVIEW

File: app/Models/User.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  public id: number
+
+  @column()
+  public userName: string
+
+  @column()
+  public email: string
+
+  @column({ serializeAs: null })
+  public password: string
```
Comment (Manuel-Antunes) [inline app/Models/User.ts:15]: Create a hook to hash this password

**Review by Manuel-Antunes — CHANGES_REQUESTED** [review]
## please solve all comments and request my review again
* Fix the password hashing issue
## Tip
* create recreate model and migration using [Adonis Auth](https://docs.adonisjs.com/guides/auth/introduction) it creates an full model with password hashing and connects into auth middleware automaticly

File: app/Models/User.ts (hunk 1 above)
Comment (ronaldemanuel) [reply app/Models/User.ts:16]: I recreated the user model using the `Adonis Auth`.

File: config/auth.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        |--------------------------------------------------------------------------
+        | Uids
+        |--------------------------------------------------------------------------
+        |
+        | Uids are used to search a user against one of the mentioned columns. During
+        | login, the auth module will search the user mentioned value against one
+        | of the mentioned columns to find their user record.
+        |
+        */
+        uids: ['email'],
```
Comment (Manuel-Antunes) [inline config/auth.ts:90]:
```suggestion
        uids: ['email', 'userName'],
```

**Review by Manuel-Antunes — APPROVED** [review]
Solve this change and merge

**Pull request merged**